src/device-api/v2.ts

```diff
@@ -42,7 +46,20 @@ export function createV2Api(router: Router, applications: ApplicationManager) {
 						res.status(404).send(appNotFoundMessage);
 						return;
 					}
-					const service = _.find(app.services, { imageId }) as Service | null;
+
+					// Work if we have a service name or an image id
+					if (imageId == null) {
+						if (serviceName == null) {
+							throw new Error(v2ServiceEndpointInputErrorMessage);
+						}
+					}
+
+					let service: Service | null;
+					if (imageId != null) {
+						service = _.find(app.services, { imageId }) as Service | null;
```

Comment:
Is `app.services` not typed?  If it is typed as `Service[]` then I would expect the ` as Service | null` to be redundant

Reply:
It is typed, but it looks like the lodash types are extremely permissive here. I tried to get away with not needing this but I was getting weird results such as `number is not assignable to Service | null`.

Reply:
If you use
```typescript
let service: Service | undefined;
if (imageId != null) {
	service = _.find(app.services, (service) => service.imageId === imageId);
} else {
	service = _.find(app.services, (service) => service.serviceName === serviceName);
}
```
then it'll work, and `Service | undefined` is the correct typing instead of `Service | null`